### ISSUE TITLE
le9ec for complex working set protection

### ISSUE THREAD

**Issue by hakavlad** [zen-kernel/zen-kernel#237]
le9db is obsoleted.

New version here: https://github.com/hakavlad/le9-patch/tree/main/le9ec_patches

What's new in le9ec patch?

- The new knob: `vm.anon_min_kbytes`;
- Improved documentation;
- Checking for the valid values of CONFIG options at the Kconfig level, not at compile time;
- The protection of the working set now takes precedence over `vm.swappiness=0`: setting swappiness=0 does not break the cache protection. Previously, setting swappiness=0 disabled the soft protection of the working set.

Recommended default values: 262144 for `CONFIG_CLEAN_LOW_KBYTES` and 0 for others.

**Comment by damentz** [comment]
Looks like we're having some strange revert/merge conflicts.  Will need to spend more time on this once I get more time.

**Comment by damentz** [comment]
I get issues like below when attempting to try any of the variations of the patch to our 5.13/lru branch.

```
Applying: mm/vmscan: add sysctl knobs for protecting the working set
error: sha1 information is lacking or useless (include/linux/mm.h).
error: could not build fake ancestor
Patch failed at 0001 mm/vmscan: add sysctl knobs for protecting the working set
hint: Use 'git am --show-current-patch=diff' to see the failed patch
When you have resolved this problem, run "git am --continue".
If you prefer to skip this patch, run "git am --skip" instead.
To restore the original branch and stop patching, run "git am --abort".
```

I'm assuming the ancestor is local to your system and without it, git can't use both `base` and `local` differences to determine how to proceed on a merge conflict.   I recommend you at least rebase your patches on top of each major version (v5.13, v5.14, etc), but pushing up a branch so we can examine the history the patch is based on is better, especially if it's something not available publicly.

Either way, we'll keep 5.13 the way it is for stability and incorporate both mgLRU and le9 into our 5.14/lru branch.  I'll keep this issue open until that's complete.

**Comment by hakavlad** [comment]
What about `patch -p1`? `le9ec-5.13-rc2-MGLRU.patch` should be correctly applied if you applied mglru patch previously.

**Comment by damentz** [comment]
le9ec is merged into 5.14/master as of https://github.com/zen-kernel/zen-kernel/commit/c8516421242f4e62300a09eca71ee0e472844e49, marking as resolved.
